### DISCUSSION TITLE
Is automatic validation of Protobuf messages in place?

### DISCUSSION THREAD

**Discussion by preslavrachev** [quarkusio/quarkus#42802] [Q&A]
Take the following Protobufs spec:

```protobuf
message SomeMessageRequest {
    string name = 1;
    optional string label = 2;
}
```

I also have a Grpc service that is suposed to accept requests of that type. A quick demo allowed me to send and empty message to that service, and it still hit the endpoint. I was expecting an error, because the `name` field is not marked as `optional`, but that wasn't the case. The response from the server was a clear 200 OK.

What is the way to validate incoming Protobuf messages in Quarkus, other than field-by-field checking?

**Comment by cescoffier** [comment]
No, there is no automatic validation. We follow the same default behavior of gRPC Java. If that's not the case anymore, please open an issue.
